### PR TITLE
Add minimum time between successive outputs

### LINKED ISSUE #23212: Allow user to specify a minimum time interval between outputs
## Reason
We can already specify sync times, but that is cumbersome for long runs. And we have simulation step intervals for output, which is fine for very homogeneous dt. _But_ a long running simulation can temporarily stagnate or go to very small dt, and we might not want to capture every time step for output. It would be nice if there was an option to say _do not output any more frequently than once (simulation time!) minute.

## Design
Add a `minimum_time_interval`  option to outputs. Store the simulation time of the last output, and skip an output if it was less than `minimum_time_interval` ago.

## Impact
Added capability

### PR DESCRIPTION
**Pull request by dschwen**
Closes #23212

### CODE REVIEW

**Review by joshuahansel — DISMISSED** [review]

**Review by GiudGiud — CHANGES_REQUESTED** [review]
I ll take a look, it s something I care about

**Review by bwspenc — DISMISSED** [review]
This looks great. This will be an incredibly helpful feature!

File: test/tests/outputs/intervals/tests [hunk 1]
```diff
@@ -29,44 +30,57 @@
...
     exodiff = 'multiple_sync_times_out_3.e multiple_sync_times_out_5.e multiple_sync_times_sync_0.e multiple_sync_times_sync_1.e'
 
-    requirement = "The system shall support multiple output objects that have different simulation synchronization times specified."
-  [../]
-  [./no_output]
+    requirement = "The system shall support multiple output objects that have different simulation "
+                  "synchronization times specified."
+  []
+  [minimum_time_interval]
+    #
```
Comment (GiudGiud) [inline test/tests/outputs/intervals/tests:51]:
```suggestion
```

File: framework/src/outputs/Output.C [hunk 2]
```diff
@@ -211,6 +220,10 @@ Output::onInterval()
   if (_sync_times.find(_time) != _sync_times.end())
     output = true;
 
+  // check if enough time has passed between outputs
+  if (_time > _last_output_time && _last_output_time + _minimum_time_interval > _time)
```
Comment (GiudGiud) [inline framework/src/outputs/Output.C:224]: use the timestep tolerance here, with a preferrence for outputting too much

**Review by GiudGiud — CHANGES_REQUESTED** [review]
Docs should make clear in what order parameters are obeyed (interval more important? min time more important?)

How is the behavior with multiple execute_on ? on LINEAR they are on the same arent they?

File: framework/src/outputs/Output.C (hunk 2 above)
Comment (dschwen) [reply framework/src/outputs/Output.C:224]: Good suggestion

**Comment by dschwen** [comment]
> Docs should make clear in what order parameters are obeyed (interval more important? min time more important?)
> 
> How is the behavior with multiple execute_on ? on LINEAR they are on the same arent they?

On linear the first condition will be false, so it will still output. That seems to be what makes most sense here.

**Comment by GiudGiud** [comment]
I think an additional paragraph in the main outputs doc could describe the rules. It could be argued either way

File: framework/doc/content/syntax/Outputs/index.md [hunk 3]
```diff
@@ -129,6 +129,16 @@ as detailed below, does not get inherited.
 Developers may refer to [CommonOutputAction.md] for more information about implementing new
 common output parameters.
 
+## Controlling output frequency
+
+Several parameters are available common to all output objects to control teh frequency with which
```
Comment (GiudGiud) [inline framework/doc/content/syntax/Outputs/index.md:134]:
```suggestion
Several parameters are available common to all output objects to control the frequency with which
```

File: framework/doc/content/syntax/Outputs/index.md [hunk 4]
```diff
@@ -129,6 +129,16 @@ as detailed below, does not get inherited.
...
+output occurs.
+
+- `interval = N` will cause output objects to outpout only every _Nth_ simulation step
+- if `start_time` and/or `end_time` are specified, outputs will only happen after the given start time and/or before the given end time
+- `sync_times = 't1 t2 t3 ... tN` will cause MOOSE to hit each listed simulation time _t1 .. tN_ exactly. With `sync_only = true` outputs will _only_ happen at the specified sync times.
+- `minimum_time_interval = dt` will suppress any output if the previous output happened at an earlier time that is more recent than _dt_ time units ago (specifically this means that outouts during linear iterations are not suppressed, as they are happening at the _same_ simulation time, and output from failed, cut steps do not lead to outout suppression in repeated steps as they happened in the future)
```
Comment (GiudGiud) [inline framework/doc/content/syntax/Outputs/index.md:140]:
```suggestion
- `minimum_time_interval = dt` will suppress any output if the previous output happened at an earlier time that is more recent than _dt_ time units ago (specifically this means that outputs during linear iterations are not suppressed, as they are happening at the _same_ simulation time, and output from failed, cut steps do not lead to output suppression in repeated steps as they happened in the future)
```

File: framework/doc/content/syntax/Outputs/index.md [hunk 5]
```diff
@@ -129,6 +129,16 @@ as detailed below, does not get inherited.
 Developers may refer to [CommonOutputAction.md] for more information about implementing new
 common output parameters.
 
+## Controlling output frequency
+
+Several parameters are available common to all output objects to control teh frequency with which
+output occurs.
+
+- `interval = N` will cause output objects to outpout only every _Nth_ simulation step
```
Comment (GiudGiud) [inline framework/doc/content/syntax/Outputs/index.md:137]:
```suggestion
- `interval = N` will cause output objects to output only every _Nth_ simulation step
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]
spelling
please squash in last commit

**Review by GiudGiud — APPROVED** [review]
nice option.
This complements nicely the step intervals method.
We re still missing a true time intervals output, maybe we ll get to it this year but this is almost it

**Pull request merged**